autoflip_calculators/text_detection_calculator.cc

```diff
@@ -0,0 +1,262 @@
+
+#include <algorithm>
+#include <memory>
+
+#include "mediapipe/examples/desktop/autoflip/autoflip_messages.pb.h"
+#include "mediapipe/examples/desktop/autoflip/quality/visual_scorer.h"
+#include "mediapipe/framework/calculator_framework.h"
+#include "mediapipe/framework/formats/detection.pb.h"
+#include "mediapipe/framework/formats/image_frame.h"
+#include "mediapipe/framework/formats/image_frame_opencv.h"
+#include "mediapipe/framework/formats/location_data.pb.h"
+#include "mediapipe/framework/port/opencv_core_inc.h"
+#include "mediapipe/framework/port/opencv_imgproc_inc.h"
+#include "mediapipe/framework/port/opencv_dnn_inc.h"
+#include "mediapipe/framework/port/ret_check.h"
+#include "mediapipe/framework/port/status.h"
+#include "mediapipe/framework/port/status_builder.h"
+
+namespace mediapipe {
+namespace autoflip {
+
+// This calculator detects texts in the images and converts detected texts 
+// to SalientRegion protos that can be used for downstream processing. Each
+// SalientRegion is scored using image cues. 
+// Example:
+//    calculator: "TextDetectionCalculator"
+//    input_stream: "VIDEO:frames"
+//    output_stream: "REGIONS:regions"
+//    options:{
+//      [mediapipe.autoflip.TextDetectionCalculatorOptions.ext]:{
+//        export_text_bounding_box: True
+//      }
+//    }
+//
+class TextDetectionCalculator : public CalculatorBase {
+ public:
+  TextDetectionCalculator();
+  ~TextDetectionCalculator() override {}
+  TextDetectionCalculator(const TextDetectionCalculator&) = delete;
+  TextDetectionCalculator& operator=(const TextDetectionCalculator&) = delete;
+
+  static ::mediapipe::Status GetContract(mediapipe::CalculatorContract* cc);
+  ::mediapipe::Status Open(mediapipe::CalculatorContext* cc) override;
+  ::mediapipe::Status Process(mediapipe::CalculatorContext* cc) override;
+
+ private:
+  double NormalizeX(const int pixel);
+  double NormalizeY(const int pixel);
+  // Decode the outputs of the EAST neural network
+  void DecodeBoundingBoxes(const cv::Mat& socres, const cv::Mat& geometry, float scoreTresh,
+                         std::vector<RotatedRect>& detections, std::vector<float>& confidences);
+
+  // Calculator options.
+  TextDetectionCalculatorOptions options_;
+
+  // A scorer used to assign weights to texts.
+  std::unique_ptr<VisualScorer> scorer_;
+
+  // Dimensions of video frame
+  int frame_width_;
+  int frame_height_;
+
+  const float confidence_threshold = 0.7; //0.5;
+  const float nms_threshold = 0.4;
+  const std::string model_path = "/mediapipe/models/frozen_east_text_detection.pb";
+  cv::dnn::Net detector;
+  std::vector<Mat> outs;
+  std::vector<String> outNames(2);
+
+}; // end with inheritance
+
+REGISTER_CALCULATOR(TextDetectionCalculator);
+
+TextDetectionCalculator::TextDetectionCalculator() {}
+
+::mediapipe::Status TextDetectionCalculator::GetContract(
+    mediapipe::CalculatorContract* cc) {
+  if (cc->Inputs().HasTag("VIDEO")) {
```

Comment:
I think you shouldn't do this check because the calculator requires video input.  Please remove the if statement.